operator/v1/types_ingress.go

```diff
@@ -558,6 +558,52 @@ const (
 	AWSNetworkLoadBalancer AWSLoadBalancerType = "NLB"
 )
 
+// AWSSubnets contains a list of references to AWS subnets by
+// ID or name.
+// +kubebuilder:validation:XValidation:rule=`has(self.ids) && has(self.names) ? size(self.ids + self.names) <= 10 : true`,message="the total number of subnets cannot exceed 10"
+// +kubebuilder:validation:XValidation:rule=`has(self.ids) || has(self.names)`,message="must specify at least 1 subnet name or id"
```

Comment:
There is a min properties validation that means you must have at least one item in the struct for it to be valid, not sure if that's better than this (since you have a custom error message), but thought I'd let you know in case you wanted to try it